tests/Unit/Utilities/Test_TupleSlice.cpp

```diff
@@ -0,0 +1,38 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <string>
+#include <tuple>
+
+#include "Utilities/TupleSlice.hpp"
+
+SPECTRE_TEST_CASE("Unit.Utilities.TupleSlice", "[Utilities][Unit]") {
+  {
+    std::tuple<int, float, double, std::string> tuple{1, 2., 3., ""};
+    CHECK(tuple_slice<1, 3>(tuple) == std::tuple<float, double>{2., 3.});
+    CHECK(tuple_slice<1, 3>(std::move(tuple)) ==
+          std::tuple<float, double>{2., 3.});
+  }
+  {
```

Comment:
In general your tests are much too simplistic. You should be testing tuples of rvalue references and lvalue references to make sure that works correctly and doesn't cause dangling references...